### PR TITLE
doc: deprecate _primary and _replica shard option

### PR DESCRIPTION
**Pull request by dnhatn**
Tells v6 users that we are going to remove these options in v7.

Related #26335

### CODE REVIEW

**Comment by dakrone** [comment]
@dnhatn we have special asciidoc syntax for deprecations, see: https://github.com/elastic/docs#additions-and-deprecations

**Comment by dnhatn** [comment]
Thanks @dakrone. I will update it.

**Review by jasontedor — CHANGES_REQUESTED** [review]
I think that we should add deprecation logging for use of these preferences.

**Comment by dnhatn** [comment]
retest this please

**Comment by dnhatn** [comment]
Thanks @jasontedor  for the suggestion. I have added deprecation logging for these parameters. Could you please have another look?

File: core/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java [hunk 1]
```diff
@@ -199,12 +199,20 @@ private ShardIterator preferenceActiveShardIterator(IndexShardRoutingTable index
                 case LOCAL:
                     return indexShard.preferNodeActiveInitializingShardsIt(Collections.singleton(localNodeId));
                 case PRIMARY:
+                    deprecationLogger.deprecated("[_primary] has been deprecated in 6.0+, and will be removed in 7.0; " +
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java]: The version here should be 6.1 (we are going to backport this to 6.x but not 6.0).

File: docs/reference/search/request/preference.asciidoc [hunk 2]
```diff
@@ -9,18 +9,21 @@ The `preference` is a query string parameter which can be set to:
 [horizontal]
 `_primary`:: 
 	The operation will go and be executed only on the primary
-	shards.
+	shards. deprecated[6.0.0, will be removed in 7.0, use `_only_nodes` or `_prefer_nodes`]
```
Comment (jasontedor) [inline docs/reference/search/request/preference.asciidoc]: This should be 6.1.0.

File: rest-api-spec/src/main/resources/rest-api-spec/test/bulk/20_list_of_strings.yml [hunk 3]
```diff
@@ -1,5 +1,10 @@
 ---
 "List of strings":
+  - skip:
+      version: " - 5.99.99"
```
Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/bulk/20_list_of_strings.yml]: Let's only skip to 6.0.99.

**Review by jasontedor — COMMENTED** [review]
I left a few more, almost there.

**Comment by jasontedor** [comment]
@dnhatn Let's be careful when we merge this that the master branch does not break because of the BWC tests that use preferences. If you need help walking through how to test that locally, please let me know.

**Review by jasontedor — APPROVED** [review]
LGTM. This looks good, but please see my note about making sure that this doesn't break master because of the BWC tests there. So: a green CI PR build on this is not enough.

**Comment by dnhatn** [comment]
@jasontedor The BWC tests are good for this branch. Thanks a lot for your guidance.

**Pull request merged**